src/Test.Utilities/DiagnosticAnalyzerTestBase.cs

```diff
@@ -395,7 +395,22 @@ protected static Project CreateProject(string[] sources, string language = Langu
             {
                 string newFileName = source.FilePath ?? fileNamePrefix + count++ + "." + fileExt;
                 DocumentId documentId = DocumentId.CreateNewId(projectId, debugName: newFileName);
-                project = project.AddDocument(newFileName, SourceText.From(source.Source)).Project;
+
+                SyntaxNode root;
+                if (language == LanguageNames.CSharp)
+                {
+                    root = CSharpSyntaxTree.ParseText(
+                        SourceText.From(source.Source),
+                        CSharpParseOptions.Default.WithLanguageVersion(Microsoft.CodeAnalysis.CSharp.LanguageVersion.Latest)).GetRoot();
```

Comment:
@333fred I made these changes because when I try to run tests I get `The name 'nameof' does not exist in the current context`; the code isn't being compiled with C# 6. This fix didn't work though. Do you know how to change the language version?

Reply:
Tagging @mavasani

Reply:
I will work on this bug today, thanks.

Reply:
@jamesqo Underlying reason for this is https://github.com/dotnet/roslyn/issues/24212. You will need to handcraft the nameof expression node instead of using SyntaxGenerator, see https://github.com/dotnet/roslyn-analyzers/pull/1525

Reply:
@mavasani Would it be OK for me to pass `allowNewCompilerDiagnostics: true, validationMode: TestValidationMode.AllowCompileErrors` instead of having to change it now and then remove the workaround later?

Reply:
@mavasani Do you know how to verify the code fix message in the unit tests? (See the outdated diff above)